circuits/kimchi/src/polynomials/lookup.rs

```diff
@@ -314,33 +313,41 @@ pub fn sorted<
     let by_row = lookup_info.by_row(gates);
     let max_lookups_per_row = lookup_info.max_per_row;
 
+    for t in lookup_table().take(lookup_rows) {
+        // Don't multiply-count duplicate values in the table, or they'll be duplicated for each
+        // duplicate! E.g. 3 values the same would insert the value 3 times for each instance.
+        counts.entry(t).or_insert(1);
+    }
+
     for (i, row) in by_row.iter().enumerate().take(lookup_rows) {
         let spec = row;
         let padding = max_lookups_per_row - spec.len();
         for joint_lookup in spec.iter() {
             let table_entry = E::evaluate(&params, joint_lookup, witness, i);
```

Comment:
off-scope for your PR, but I think we ought to specify what all these terms mean. Here `table_entry` makes you think that it's an entry in the table, but it's not, it's a joint tuple of values you want to lookup in the table.

Reply:
I've renamed `table_entry` to `joint_lookup_evaluation`. Hopefully that's clearer?